api/server/httputils/errors.go

```diff
@@ -48,7 +49,19 @@ func GetHTTPErrorStatusCode(err error) int {
 		// there are errors falling back into this logic.
 		// If we need to differentiate between different possible error types,
 		// we should create appropriate error types that implement the httpStatusError interface.
+
 		errStr := strings.ToLower(errMsg)
+		// if the error has a type of grpc.rpcError, the error message should
+		// contain a sub string of "rpc error: code = "
+		if strings.Contains(errStr, "rpc error: code = ") {
```

Comment:
Thanks for your feedback. @cpuguy83   I think what you said is reasonable. Thanks a lot.
While I think CI fails according to accident. Can you re-trigger that? @thaJeztah